test/implementations/test_mct_barenco_74_dirty.py

```diff
@@ -56,6 +57,14 @@ def _take_auxiliaries_no(self, controls_no: int):
 
         return self._auxiliary_dict[controls_no]
 
+    def test_init(self):
+        with pytest.raises(Exception) as e_info:
+            MCTBarenco74Dirty(2)
+
+        assert e_info.value.args[0] == "Number of controls must be >= 5 for this implementation"
+
+        assert MCTBarenco74Dirty(5)

```

Comment:
I found this a little better style

Reply:
The thing is it is not exactly obvious what is tested

Reply:
so, what should I add for line 66?

Reply:
for example:
```
try:
    MCTBarenco74Dirty(5)
except e:
    assert False, "object MCTBarenco74Dirty(5) was not created, but it should be"
 ```

Reply:
fixed with the last commit